go/stellar/stellarsvc/anchor.go

```diff
@@ -0,0 +1,233 @@
+package stellarsvc
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/stellar1"
+	"golang.org/x/net/publicsuffix"
+)
+
+// anchorInteractor is used to interact with the sep6 transfer server for an asset.
+type anchorInteractor struct {
+	accountID     stellar1.AccountID
+	asset         stellar1.Asset
+	httpGetClient func(mctx libkb.MetaContext, url string) (code int, body []byte, err error)
+}
+
+// newAnchorInteractor creates an anchorInteractor for an account to interact
+// with an asset.
+func newAnchorInteractor(accountID stellar1.AccountID, asset stellar1.Asset) *anchorInteractor {
+	return &anchorInteractor{
+		accountID:     accountID,
+		asset:         asset,
+		httpGetClient: httpGet,
+	}
+}
+
+// Deposit runs the deposit action for accountID on the transfer server for asset.
+func (a *anchorInteractor) Deposit(mctx libkb.MetaContext) (stellar1.AssetActionResultLocal, error) {
+	if err := a.checkAsset(mctx); err != nil {
+		return stellar1.AssetActionResultLocal{}, err
+	}
+	u, err := a.checkURL(mctx, "deposit")
+	if err != nil {
+		return stellar1.AssetActionResultLocal{}, err
+	}
+
+	v := url.Values{}
+	v.Set("account", a.accountID.String())
+	v.Set("asset_code", a.asset.Code)
+	u.RawQuery = v.Encode()
+
+	var okResponse okDepositResponse
+	return a.get(mctx, u, &okResponse)
+}
+
+// Withdraw runs the withdraw action for accountID on the transfer server for asset.
+func (a *anchorInteractor) Withdraw(mctx libkb.MetaContext) (stellar1.AssetActionResultLocal, error) {
+	if err := a.checkAsset(mctx); err != nil {
+		return stellar1.AssetActionResultLocal{}, err
+	}
+	u, err := a.checkURL(mctx, "withdraw")
+	if err != nil {
+		return stellar1.AssetActionResultLocal{}, err
+	}
+
+	// TODO: `type` is supposedly optional, but actually required...so figure it out.
```

Comment:
on https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0006.md it looks like it's required for withdrawals and optional for deposits. i dont understand though. it doesn't look like we're including it. and this says account is required for withdraws, but it doesn't look like we're sending that either.

Reply:
I asked and was told it was optional several days ago.  There is a PR to change the spec:

https://github.com/stellar/stellar-protocol/pull/358

but all the anchors i've tried currently require it.

i'm going to fix this now by getting additional info from stellard, but i wanted to get the deposit flow in for UI testing.

Reply:
ahh ahh this is what you were saying earlier. got it.